### PR TITLE
(WIP) add some dilations

### PR DESCRIPTION
**Pull request by b-mehta**
I added the definition of dilation and the properties about it, I couldn't get `quux` to work though - it might be because I've screwed up the definition of dilation somehow, but it's not immediately obvious to me why it's true. Though, I generated some center mass lemmas which I think are missing from mathlib. I also added barycenter and affine independence, plus the proof that a linearly independent family is affinely independent.

### CODE REVIEW

File: src/local/int_cvx.lean [hunk 1]
```diff
@@ -16,24 +18,76 @@ import analysis.convex.caratheodory
 variables {V : Type*} [add_comm_group V] [vector_space ℝ V]
 
 /-- The dilation about `c` with scale factor `r`. -/
-def dilation (c : V) (r : ℝ) : V → V := sorry
+def dilation (c : V) (r : ℝ) : V → V := λ v, r • (v - c) + c
```
Comment (jcommelin) [inline src/local/int_cvx.lean:21]: Isn't this something that should be done more generally for affine spaces?

File: src/local/int_cvx.lean [hunk 2]
```diff
@@ -68,8 +131,19 @@ lemma quux' (x : V) (t : finset V) (f : V → ℝ) (w : t.sum f = 1) (h : t.cent
   t.center_mass (λ x, r⁻¹ * f x + (1 - r⁻¹) / (t.card)) (λ x, dilation (barycenter t) r x) = x :=
 sorry
 
-def affine_independent (ι : Type*) (f : ι → V) : Prop := sorry
-def set.affine_independent (s : set V) : Prop := sorry
+def affine_independent (ι : Type*) (f : ι → V) : Prop :=
```
Comment (jcommelin) [inline src/local/int_cvx.lean:134]: This is now in mathlib, right?

**Comment by PatrickMassot** [comment]
Thanks, but I think you missed stuff that happened on mathlib's side, like https://github.com/leanprover-community/mathlib/blob/46a889442d748dcb76c08b1732feaf313db83ca0/src/linear_algebra/affine_space.lean#L616 and https://github.com/leanprover-community/mathlib/pull/3140. Could you please review these and adjust your work if needed?

**Comment by b-mehta** [comment]
Oh, I didn't know that. I'll amend, thanks.

**Comment by PatrickMassot** [comment]
I'm closing this since it has been abandoned and is no longer relevant, but your contributions to the project are still welcome!

**Pull request closed without merging**